app/models/fact_value.rb

```diff
@@ -9,7 +9,7 @@ class FactValue < ActiveRecord::Base
 
   has_one :parent_fact_name, :through => :fact_name, :source => :parent
 
-  scoped_search :on => :value, :in_key=> :fact_name, :on_key=> :name, :rename => :facts, :complete_value => true, :only_explicit => true
+  scoped_search :on => :value, :in_key=> :fact_name, :on_key=> :name, :rename => :facts, :complete_value => true, :only_explicit => true, :ext_method => :search_cast_facts
   scoped_search :on => :value, :default_order => true
```

Comment:
> Oh, because the name is optional then, hm. I suppose that'd work if that condition was in the search_cast_facts methods instead.

How about
```
      limit_name = ((key.include? '.') ? "fact_names.name = '#{key.split('.')[1]}' AND" : '')
...
      { :conditions => "#{limit_name} #{casted}"}
```